packages/core/src/agent/context/AgentContext.ts

```diff
@@ -0,0 +1,45 @@
+import type { DependencyManager } from '../../plugins'
+import type { Wallet } from '../../wallet'
+
+import { InjectionSymbols } from '../../constants'
+import { AgentConfig } from '../AgentConfig'
+
+export class AgentContext {
+  /**
+   * Dependency manager holds all dependencies for the current context. Possibly a child of a parent dependency manager,
+   * in which case all singleton dependencies from the parent context are also available to this context.
+   */
+  public readonly dependencyManager: DependencyManager
+
+  /**
+   * An identifier that allows to correlate this context across usages. An example of the contextCorrelationId could be
+   * the id of the `TenantRecord` that is associated with this context. The AgentContextProvider can use this identifier to
+   * correlate an inbound message to a specific context (if the message is not encrypted, it's impossible to correlate it to a tenant)
```

Comment:
`(if the message is not encrypted, it's impossible to correlate it to a tenant)`

Could you eloborate on this a bit more, why not?

Reply:
If we have a plaintext didcomm message like the following it's impossible to know which tenant this message is for. There's no information that correlates this to a specific tenant. 

```json
{
    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/issue-credential/1.0/propose-credential",
    "@id": "<uuid-of-propose-message>",
    "comment": "some comment",
    "credential_proposal": <json-ld object>,
    "schema_id": "Schema ID string",
    "cred_def_id": "Credential Definition ID string"
}
```

If we have an encrypted message however, we can look at the recipient public keys of the messages. We have a mapping internally that registers which public keys belong to which tenant so we can route the message to the correct tenant.

In cases of out of band messages we often already know which context we're in because we receive it through the agent API, not through the inbound transport, so that allows us to just pass the contextCorrelationId

Reply:
Got it. That makes sense to me.